odoo/addons/test_new_api/tests/test_new_fields.py

```diff
@@ -1603,6 +1603,26 @@ def test_60_x2many_domain(self):
         self.assertFalse(discussion.very_important_messages)
         self.assertFalse(message.exists())
 
+    def test_60_many2many_domain(self):
+        """ test the cache consistency of a many2many field with a domain """
+        discussion = self.env.ref('test_new_api.discussion_0')
+        category = self.env['test_new_api.category'].create({'name': "Foo"})
+        discussion.categories = category
+        discussion.flush()
+        discussion.invalidate_cache()
+
+        # patch the many2many field to give it a domain
+        field = discussion._fields['categories']
+        self.patch(field, 'domain', [('color', '!=', 42)])
+        self.registry.setup_models(self.cr)
+
+        # the category is in the many2many
+        self.assertIn(category, discussion.categories)
+
+        # modify the category; it should not longer be in the many2many
+        category.color = 42
+        self.assertNotIn(category, discussion.categories)
```

Comment:
do you think it's worth it to test the opposite too? changing a record of the co-model so that it goes from filtered to unfiltered? in theory it's the same codepath but idk, we could add:

```python
category.color = 69
self.assertIn(category, discussion.categories)
```

just to be complete, I leave it up to you